uaparser/parser.go

```diff
@@ -202,6 +208,7 @@ func New(regexFile string) (*Parser, error) {
 	}
 	matchIdxNotOk = cDefaultMatchIdxNotOk
 	missesTreshold = cDefaultMissesTreshold
+	parserCacheSize = cDefaultCacheSize
 	parser, err := NewFromBytes(data)
```

Comment:
```suggestion
	parser, err := newFromBytes(data, cDefaultCacheSize)
```

I don't love that these functions are using a module-level variables to communicate extra values to NewFromBytes; feels like it should just be another parameter.  I think to achieve that without changing the api we might need to introduce a shared helper function for the current reuse of the public functions - e.g. NewWithOptions could call a newFromBytes that's an internal helper that takes the cache size as a parameter.  (Fwiw it looks like NewFromBytes was intended to be an internal function, but not sure it makes sense to just change that - safer to make it call a shared helper and always pass the default cache size)

The main downside of the module-level variable, other than code cleanliness, is if there are for some reason multiple goroutines or threads creating Parser objects at the same time, they could have a data race on parserCacheSize (or on missesTreshold / matchIdxNotOk) so it'd be better to avoid that possibility.

Reply:
> I don't love that these functions are using a module-level variables to communicate extra values to NewFromBytes

I don't like it either, but I followed the approach that was taken before. And yes, it causes a race condition, plus you can't create two parsers with different options at the same time because of these global variables.

> I think to achieve that without changing the api we might need to introduce a shared helper function for the current reuse of the public functions

My point about the API change was about the `NewWithOptions` function: I introduced another parameter, which essentially changed the function declaration and broke backward compatibility. A better solution, in my opinion, is to use some `Config` structure, which contains all the necessary configuration options, and you can extend it in the future if needed.

Reply:
I've implemented this idea, please take a look.